### DISCUSSION TITLE
New navigation primitive: `UISplitViewController`

### DISCUSSION THREAD

**Discussion by okwasniewski** [software-mansion/react-native-screens#2026] [Ideas]
Hey! 

I propose a new navigator type for `@react-navigation/native` and `react-native-screens` that backs it up. The idea is to have a Split-view navigation for larger devices that has a list on the left side and a detail view on right. 

![image](https://github.com/software-mansion/react-native-screens/assets/52801365/4512ca8d-fb86-4cac-ae8c-e613f6fb23bd)

This design would benefit apps written for iPad and visionOS and also for macOS through MacCatalyst.

The native settings app is a perfect representation of this pattern:

## Screenshots 
![CleanShot 2024-02-02 at 11 32 30@2x](https://github.com/software-mansion/react-native-screens/assets/52801365/34aea643-5079-4b72-b010-0c777d15f01c)

![image](https://github.com/software-mansion/react-native-screens/assets/52801365/582ae6d0-3ac9-4f3b-bdef-c629294f3047)

## Resources
- https://developer.apple.com/documentation/uikit/uisplitviewcontroller
- https://developer.apple.com/tutorials/mac-catalyst/displaying-the-sidebar

Let me know what do you think 👍🏻

**Comment by tboba** [comment]
Hey @okwasniewski, thanks for starting this discussion!
I think that it's a very very great idea to have that in screens! There are already plans to support `UISplitViewController`. We have that point in our agenda 💪

**Reply by elsa17z** [reply]
May I ask what is the roadmap on this or is this still too early to say? Thanks a lot for your work!

**Reply by tboba** [reply]
@elsa17z Hi hi, sorry for late response on this. Unfortunately I'm not an active developer of screens right now, however this feature is still waiting on PR https://github.com/software-mansion/react-native-screens/pull/2171 - right now, we've got prepared a PoC of working split screen support, however it's still not ready to use in RN apps. We'll keep you updated if something will change in this topic 🙏

**Comment by erickreutz** [comment]
Did this ever land anywhere?

**Reply by tboba** [reply]
Hi @erickreutz, some time ago I created a PR, dedicated for adding UISplitViewController to screens -https://github.com/software-mansion/react-native-screens/pull/2171. Unfortunately, I'm not developing screens actively for now, but I believe this feature is still on screens roadmap!
